radix-engine/src/model/transaction_processor/executables.rs

```diff
@@ -251,6 +256,11 @@ impl<'a> Executor for TransactionProcessorRunInvocation<'a> {
             RENode::TransactionRuntime(runtime_substate),
         )?;
 
+        let mut blobs_by_hash = HashMap::new();
+        for blob in self.blobs.as_ref() {
+            blobs_by_hash.insert(hash(blob), blob);
```

Comment:
Hashing's quite expensive - what are your thoughts on preventing any blob hashing / blob commands till after the fee loan has been paid back? (and calculating the hashes lazily on first use).

Not in this PR necessarily 👍

Reply:
That said, I guess we don't delay sig verification... although hashing 1MB is possibly slower.

Reply:
That's a good callout. Let me add a TODO comment here.